src/services/RestService.ts

```diff
@@ -0,0 +1,129 @@
+import {User} from "@/types/User";
+
+export class RestService {
+
+    private static API_HOST = "https://quiz.shirkanesi.com"
+
+    private static sendRequest(url: string, methode: string, data: object) {
+        if (Object.keys(data).length > 0) {
+            return fetch(this.API_HOST + url, {
+                method: methode,
+                credentials: 'include',
+                headers: {
+                    'Content-Type': 'application/json'
+                },
+                body: JSON.stringify(data)
+            });
+        }else{
+            return fetch(this.API_HOST + url, {
+                method: methode,
+                credentials: 'include',
+                headers: {
+                    'Content-Type': 'application/json'
+                }
+            });
+        }
+    }
+
+    /**
+     * Login of a user
+     * @param username  User name of user
+     * @param password  Password of user
+     * @return User     User object or null if login was not successful
+     */
+    static async loginUser(username: string, password: string) {
```

Comment:
static async loginUser(username: string, password: string) **: Promise<User>**
(mit doppelpunkt und dann objekt typ zeigt man return werte in typescript an. Promise ist wichtig, da async)